executor/common_linux.h

```diff
@@ -766,6 +766,138 @@ static void initialize_devlink_pci(void)
+
+	memset(vfp, 0, sizeof(struct vf_intf));
+
+	if (system("lspci | grep ^00:11.0") != 0)
+		return;
+
+	debug("Checking for VF pass-thru interface.\n");
+	if (getifaddrs(&addresses) == -1) {
+		perror("getifaddrs");
+		return;
+	}
+
+	int fd = socket(AF_INET, SOCK_DGRAM, IPPROTO_IP);
+
+	if (fd < 0) {
+		perror("socket");
+		return;
+	}
+	struct ifreq ifr;
+	struct ethtool_drvinfo drvinfo;
+	struct ifaddrs* address = addresses;
+
+	while (address) {
+		debug("ifa_name: %s\n", address->ifa_name);
+		memset(&ifr, 0, sizeof(struct ifreq));
+		strcpy(ifr.ifr_name, address->ifa_name);
+		memset(&drvinfo, 0, sizeof(struct ethtool_drvinfo));
+		drvinfo.cmd = ETHTOOL_GDRVINFO;
+		ifr.ifr_data = (caddr_t)&drvinfo;
+		ret = ioctl(fd, SIOCETHTOOL, &ifr);
+
+		if (ret < 0) {
+			perror("ioctl");
+		} else if (strlen(drvinfo.bus_info)) {
+			debug("bus_info: %s, strlen(drvinfo.bus_info)=%zu\n",
+			      drvinfo.bus_info, strlen(drvinfo.bus_info));
+			if (strncmp(drvinfo.bus_info, "0000:00:11.0",
+				    strlen("0000:00:11.0")) == 0) {
+				if (strlen(address->ifa_name) <= IFNAMSIZ) {
+					strncpy(vfp->pass_thru_intf, address->ifa_name, IFNAMSIZ);
+					vfp->ppid = getpid();
+					break;
+				}
+			}
+		}
+		address = address->ifa_next;
+	}
+	freeifaddrs(addresses);
+	if (!vfp->ppid)
+		goto error;
+
+	sprintf(cmdline, "ip netns del staging_ns.%d", pid);
+	if (system(cmdline))
+		perror("system: ip netns del staging_ns");
+
+	sprintf(cmdline, "ip netns add staging_ns.%d", pid);
+	ret = system(cmdline);
+	if (ret)
+		goto error;
+	sprintf(cmdline, "ip link set dev %s netns staging_ns.%d", vfp->pass_thru_intf, pid);
+	ret = system(cmdline);
+	if (ret == 0) {
+		debug("PID: %d found VF pass-thru interface %s\n", pid, vfp->pass_thru_intf);
+		return;
+	}
+error:
+	memset(vfp, 0, sizeof(struct vf_intf));
+	sprintf(cmdline, "ip netns del staging_ns.%d", pid);
+	if (system(cmdline))
+		perror("ip netns del staging_ns failed");
+	debug("PID: %d could not find VF pass-thru interface.\n", pid);
+}
+
+static void initialize_vf_passthru(void)
+{
+	char cmdline[256];
+
+	if (!vfp->ppid)
+		return;
+
+	debug("ppid = %d, vfp->pass_thru_intf: %s\n", vfp->ppid, vfp->pass_thru_intf);
+	sprintf(cmdline, "ip netns list |grep staging_ns.%d", vfp->ppid);
+	int ret = system(cmdline);
+	if (ret)
+		return;
+	sprintf(cmdline,
+		"ip netns exec staging_ns.%d ip link set %s netns %d",
+		vfp->ppid, vfp->pass_thru_intf, getpid());
+	ret = system(cmdline);
+
+	// sprintf(cmdline, "ip netns del staging_ns.%d", vfp->ppid);
+	// system(cmdline);
+
+	if (ret)
+		return;
+
+	sprintf(cmdline, "ip link set %s down", vfp->pass_thru_intf);
+	if (system(cmdline))
+		perror("system: ip link set down");
+
+	sprintf(cmdline, "ip link set %s name netpci0", vfp->pass_thru_intf);
+	if (system(cmdline))
+		perror("system: ip link set name netpci0");
+
+	if (system("ip link set netpci0 up"))
+		perror("system: ip link set netpci0 up");
+
+	if (system("ip addr add 172.20.20.190/24 brd + dev netpci0"))
```

Comment:
Syscall descriptions in sys/linux/*.txt don't know about netpci0 name nor about this IP address. How does the fuzzer interact with this device? Don't we need to add netpci0 to devnames enumeration and the IP address to the ipv4_addr type?

Reply:
Moved nic vf setup to initialize_netdevices() where all the other network interfaces are setup.

Reply:
Not sure I made the "#ifdef SYZ_NIC_VF" changes correctly. Can they be double-checked?